### PR TITLE
Apply new badge styles to en locale for designSetup step

### PR DESCRIPTION
**Pull request by candy02058912**
Related to https://github.com/Automattic/wp-calypso/issues/97717 and https://github.com/Automattic/wp-calypso/issues/97686

## Proposed Changes

> [!NOTE]  
> New badge styles are only on locale `en` for code simplicity until we get the latest translations.

* Apply the new badge style updates to design picker in onboarding flows for locale `en`
* Allow /setup/onboarding with `flags=onboarding/goals-first` to continue to see the changes (not released to public yet)
* Use `hasEnTranslation` check while we wait for translations to complete because `designSetup` is in production for other onboarding-related flows
* Added a shorter copy for when the badges overflow

badge long@2x

## Testing Instructions

* On en locale, go to /setup/onboarding
* After getting to the launchpad, go to the select a design step
* It should reflect the new badge changes
* Make sure nothing changes in the theme showcase /themes

| Old | New (en locale) |
|--------|--------|
|  |  |
|  |  | 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by madhusudhand — COMMENTED** [review]
It works well in my testing!

Small UI overlap on medium screens about `790px`. Also, they are not vertically centered.

image

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-is-updated-badge-design.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { useMemo } from 'react';
+import { useGoalsFirstExperiment } from 'calypso/landing/stepper/declarative-flow/helpers/use-goals-first-experiment';
+import { getStepFromURL } from 'calypso/landing/stepper/utils/get-flow-from-url';
+
+const useIsUpdatedBadgeDesign = () => {
+	// TODO: Remove the locale and isGoalsAtFrontExperiment check after translations are complete.
```
Comment (zaguiini) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-is-updated-badge-design.tsx]: We can use the `hasTranslation` helper, which would remove some of the time pressure to revert these changes. WDYT?

**Review by zaguiini — APPROVED** [review]
Works fine for me; left an implementation recommendation/question.

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-is-updated-badge-design.tsx (hunk 1 above)
Comment (candy02058912) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-is-updated-badge-design.tsx]: I checked the translations today, and they're deployed so we're good to go now 👍 Will update the PR

Comment (candy02058912) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-is-updated-badge-design.tsx]: I applied hasEnTranslation after adding in a new string today https://github.com/Automattic/wp-calypso/pull/97734/commits/770be1e316797f0760d29b0006204f0950be520f

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17076603

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @candy02058912 for including a screenshot in the description! This is really helpful for our translators.

**Comment by candy02058912** [comment]
> It works well in my testing!
> 
> Small UI overlap on medium screens about `790px`. Also, they are not vertically centered.
> 

Addressed the issue https://github.com/Automattic/wp-calypso/pull/97734/commits/8c920abc2f61c397839017e9ea4489998ff4ca43
image

**Comment by candy02058912** [comment]
Also added an ellipsis hint for themes that are too long, when hovering on it, the title will render

### 1. Onboarding design picker
new badge@2x

### 2. Logged-in theme showcase design picker
logged in@2x

### 3. Logged-out design picker
logged out@2x

cc: @Automattic/lego @Automattic/t-rex teams more related to the theme's code in case there are areas I have missed tested.

**Review by zaguiini — APPROVED** [review]

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17076603).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17076603).
